src/index.jsx

```diff
@@ -78,24 +108,30 @@ class CircularColor extends PureComponent {
       const y = evt.clientY - yBlock - size / 2; 
 
       const angle = -Math.atan2(y, x);
-      const handlerSize = Math.floor(size / 2 * 0.3 / 2);
 
-      this.setState({
-        cx: Math.floor((size / 2) + (Math.floor(size / 2 * 0.6) + handlerSize) * Math.cos(angle)),
-        cy: Math.floor((size / 2) - (Math.floor(size / 2 * 0.6) + handlerSize) * Math.sin(angle)),
-        color: hsvToRgb(Math.abs(angle) * 180 / Math.PI)
-      });
+      this.changeStates(angle);
+    }
+  }
 
-      if(typeof onChange === 'function') {
-        onChange(hsvToRgb(Math.abs(angle) * 180 / Math.PI));
-      }
+  changeStates = (angle) => {
```

Comment:
As it seems to me, the better name of this method is something like "changeHandlerPosition"

Reply:
Or maybe "updateHandlerPosition"

Reply:
Just inside this function, we change the handler position as well as the color. Moreover, we call the callback from the parent component. Would it be better to divide this function into two different functions?

Reply:
Actually, I don't think so. In this case, it's overhead